packages/flutter/lib/src/gestures/hit_test.dart

```diff
@@ -73,12 +73,59 @@ class HitTestEntry {
   Matrix4 _transform;
 }
 
+class _Ref<T> {
+  _Ref(this.value);
+
+  T value;
+}
+
+// A type of data that can be applied to a matrix by left-multiplication.
+@immutable
+abstract class _TransformPart {
+  const _TransformPart();
+
+  factory _TransformPart.matrix(Matrix4 matrix) => _MatrixTransformPart(matrix);
```

Comment:
Because the matrix transform part is special: It is the only kind that we can extract a matrix out of. This is important because ultimately we will need to extract a matrix to get the transform matrix.

Reply:
Removed.